pkg/productinfo/alibaba/productinfo_alibaba.go

```diff
@@ -0,0 +1,207 @@
+func NewAlibabaInfoer(regionId, accessKeyId, accessKeySecret string) (*AlibabaInfoer, error) {
+
+	// Create an ECS client
+	ecsClient, err := ecs.NewClientWithAccessKey(
+		regionId,
+		accessKeyId,
+		accessKeySecret,
+	)
+	if err != nil {
+		// Handle exceptions
+		panic(err)
+	}
+
+	return &AlibabaInfoer{
+		client: ecsClient,
+	}, nil
+}
+
+// Initialize downloads and parses the ECS price list on Alibaba Cloud
+func (e *AlibabaInfoer) Initialize() (map[string]map[string]productinfo.Price, error) {
+	log.Debug("initializing Alibaba price info")
+
+	var allPrices = struct {
+		sync.RWMutex
+		prices map[string]map[string]productinfo.Price
+	}{prices: make(map[string]map[string]productinfo.Price)}
+	var waitGroup sync.WaitGroup
+	zonesInRegions := make(map[string][]string)
+
+	regions, err := e.GetRegions()
+	if err != nil {
+		return nil, err
+	}
+
+	req := ecs.CreateDescribeInstanceTypesRequest()
+	req.RegionId = "eu-central-1"
+
+	vmSizes, err := e.client.DescribeInstanceTypes(req)
+	if err != nil {
+		return nil, err
+	}
+	instanceTypes := vmSizes.InstanceTypes.InstanceType
+	for region := range regions {
+		zones, err := e.GetZones(region)
+		if err != nil {
+			return nil, err
+		}
+		zonesInRegions[region] = zones
+		waitGroup.Add(1)
+		go func(region string, instanceTypes []ecs.InstanceType, zonesInRegions map[string][]string) {
+			defer waitGroup.Done()
+			for _, instanceType := range instanceTypes {
+				request := ecs.CreateDescribeSpotPriceHistoryRequest()
+				request.RegionId = region
+				request.NetworkType = "vpc"
+				request.InstanceType = instanceType.InstanceTypeId
+				request.OSType = "linux"
+				allPrices.RLock()
+				if allPrices.prices[region] == nil {
+					allPrices.RUnlock()
+					allPrices.Lock()
+					allPrices.prices[region] = make(map[string]productinfo.Price)
+					allPrices.Unlock()
+					allPrices.RLock()
+				}
+				allPrices.RUnlock()
+
+				prices, err := e.client.DescribeSpotPriceHistory(request)
+				if err != nil {
+					return
+				}
+
+				allPrices.RLock()
+				price := allPrices.prices[region][instanceType.InstanceTypeId]
+				allPrices.RUnlock()
+				spotPrice := make(productinfo.SpotPriceInfo)
+				priceTypes := prices.SpotPrices.SpotPriceType
+				for _, priceType := range priceTypes {
+					price.OnDemandPrice = priceType.OriginPrice
+					for _, z := range zonesInRegions[region] {
+						if z == priceType.ZoneId {
+							spotPrice[z] = priceType.SpotPrice
+							price.SpotPrice = spotPrice
+							break
+						}
+					}
+
+					allPrices.Lock()
+					allPrices.prices[region][priceType.InstanceType] = price
+					allPrices.Unlock()
+				}
+			}
+		}(region, instanceTypes, zonesInRegions)
+	}
+	waitGroup.Wait()
+
+	log.Debug("finished initializing Alibaba price info")
+	return allPrices.prices, nil
+}
+
+// GetAttributeValues gets the AttributeValues for the given attribute name
+func (e *AlibabaInfoer) GetAttributeValues(attribute string) (productinfo.AttrValues, error) {
```

Comment:
please implement this API call; it's used by the productinfo clients (eg. telescopes)